File: examples_py/helloworld/subscriber.py
import os
import sys

sys.path.insert(
    0,
    os.path.join(
        os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))),
        "libs",
        "python",
    ),
)

import zv_robot_sdk_python as zv
import time

print(f"zv_robot_sdk_python version: {zv.__version__}")

HELLO_WORLD_TOPIC = "rt/chatter"


def Handler(hello_world_msg: zv.String_):
    print(f"message:{hello_world_msg.data_}")


zv.ChannelFactory.Instance().Init(0)
subscriber = zv.ChannelSubscriber_String_(HELLO_WORLD_TOPIC)
subscriber.InitChannel(Handler)
while True:
    time.sleep(10)

File: examples/helloworld/subscriber.cpp
#include <thread>

#include <zv/robot/channel/channel_subscriber.hpp>
#include "String_.hpp"

static const std::string HELLO_WORLD_TOPIC = "rt/chatter";   // ros2 run demo_nodes_cpp talker 

using namespace zv::robot;
using namespace zv::common;

void Handler(const void* msg)
{
    const std_msgs::msg::dds_::String_* hello_world_msg = (const std_msgs::msg::dds_::String_*)msg;

    std::cout << "message:" << hello_world_msg->data_() << std::endl;
}

int main()
{
    ChannelFactory::Instance()->Init(0);
    ChannelSubscriber<std_msgs::msg::dds_::String_> subscriber(HELLO_WORLD_TOPIC);
    subscriber.InitChannel(Handler);

    while (true)
    {
        std::this_thread::sleep_for(std::chrono::seconds(10));
    }

    return 0;
}
